modules/internal/ChapelLocale.chpl

```diff
@@ -202,28 +202,35 @@ module ChapelLocale {
   /*
     Get the unique integer identifier for this locale.
 
-  :returns: index of this locale in the range ``0..numLocales-1``
-  :rtype: int
+    :returns: index of this locale in the range ``0..numLocales-1``
+    :rtype: int
 
   */
   inline proc locale.id: int {
     return this._value.id;
   }
 
   /*
-    This is the maximum task concurrency that one can expect to
-    achieve on this locale.  The value is an estimate by the
-    runtime tasking layer.  Typically it is the number of physical
-    processor cores available to the program.  Creating more tasks
-    than this will probably increase walltime rather than decrease
-    it.
+    Get the maximum task concurrency that one can expect to
+    achieve on this locale.
+
+    :returns: the maximum number of tasks that can run in parallel
+      on this locale
+    :rtype: int
+
+    Note that the value is an estimate by the runtime tasking layer.
+    Typically it is the number of physical processor cores available
+    to the program.  Creating more tasks than this will probably increase
+    walltime rather than decrease it.
```

Comment:
I know you did not add this, however I question whether we should make this kind of statement. A Chapel program should be free to create as many tasks as it deems necessary and the performance should not be penalized by doing so. What do you think about removing this last statement?

Reply:
I'm not totally sure. 

I think what this note is getting at, is that executing a single parallel construct with more tasks than the number of available cpu-threads will cause some tasks to have to wait, which is sub-optimal for performance. (i.e. the most efficient thing to do, is to run parallel loops with as many tasks as there are cpu cores, so that all tasks can run simultaneously).

Regaurdless, I suppose the wording could be improved. Maybe saying "creating more tasks than this <ins>in a single parallel construct</ins> will probably increase walltime rather than decrease it." would be more clear?

Reply:
In my view this sentence just does not add any value. Isn't its intended meaning a direct consequence of the previous sentence?

If you think we should keep it, I suggest we rephrase it for data-parallel computations. For example "Executing a data-parallel construct with more tasks than that is unlikely to improve performance."

Reply:
Ok, I'm happy with that change. I'll implement it and merge.